src/DotLiquid.Tests/Util/NumericConverterTests.cs

```diff
@@ -0,0 +1,177 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using DotLiquid.Util;
+using NUnit.Framework;
+
+namespace DotLiquid.Tests.Util
+{
+    [TestFixture]
+    public class NumericConverterTests
+    {
+        [Test]
+        public void TestCoerceToReal()
+        {
+            Assert.That(15.CoerceToReal(CultureInfo.InvariantCulture, 0), Is.EqualTo(15m).And.TypeOf(typeof(decimal)));
+            Assert.That(15m.CoerceToReal(CultureInfo.InvariantCulture, 0), Is.EqualTo(15m).And.TypeOf(typeof(decimal)));
+            Assert.That(15.0f.CoerceToReal(CultureInfo.InvariantCulture, 0), Is.EqualTo(15).And.TypeOf(typeof(double)));
+            Assert.That(15.0.CoerceToReal(CultureInfo.InvariantCulture, 0), Is.EqualTo(15).And.TypeOf(typeof(double)));
+            Assert.That("15".CoerceToReal(CultureInfo.InvariantCulture, 0), Is.EqualTo(15m).And.TypeOf(typeof(decimal)));
+            Assert.That("-15".CoerceToReal(CultureInfo.InvariantCulture, 0), Is.EqualTo(-15m).And.TypeOf(typeof(decimal)));
+            Assert.That("15.0".CoerceToReal(CultureInfo.InvariantCulture, 0), Is.EqualTo(15m).And.TypeOf(typeof(decimal)));
+        }
+
+        [Test]
+        public void TestCoerceToRealOverflow()
+        {
+            double largeValue = double.Parse("1e203");

```

Comment:
consider using `var` for obvious types, e.g.

```suggestion
            var largeValue = double.Parse("1e203");
```

Reply:
Deleted.